misk-hibernate/src/main/kotlin/misk/vitess/StartVitessService.kt

```diff
@@ -347,9 +343,41 @@ class StartVitessService(
 
   fun cluster() = cluster!!.cluster
 
+  fun shouldRunVitess() =  config.type != DataSourceType.VITESS || environment == TESTING || environment == DEVELOPMENT
```

Comment:
`fun shouldRunVitess() =  config.type != DataSourceType.VITESS ` is confusing imo. I'm not sure why `onfig.type != DataSourceType.VITESS` should possibly pull the docker image

Reply:
re: docker image, it's because it takes like 5 minutes to pull and service startup will usually time out before that. There isn't a better execution path at the moment to pull the image

Reply:
eh it's a bug. unfortunately tests aren't going to catch it because they are always going to run it in TESTING :) thanks for catching

Reply:
shouldn't this be `config.type == DataSourceType.VITESS && (environment == TESTING || environment == DEVELOPMENT)`?

Reply:
lol yes